src/audio/mixer/mixer_hifi.c

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Andrula Song <xiaoyuan.song@intel.com>
+
+#include <sof/audio/mixer.h>
+#include <sof/common.h>
+
+#if __XCC__ && (XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4)
+
+#include <xtensa/tie/xt_hifi3.h>
+
+#if CONFIG_FORMAT_S16LE
+/* Mix n 16 bit PCM source streams to one sink stream */
+static void mix_n_s16(struct comp_dev *dev, struct audio_stream __sparse_cache *sink,
+		      const struct audio_stream __sparse_cache **sources, uint32_t num_sources,
+		      uint32_t frames)
+{
+	ae_int16x4 * in[PLATFORM_MAX_CHANNELS];
+	ae_int16x4 *out = sink->w_ptr;
+	ae_int16x4 sample = AE_ZERO16();
+	ae_int16x4 res = AE_ZERO16();
+	ae_int32x2 val1 = AE_ZERO32();
+	ae_int32x2 val2 = AE_ZERO32();
+	ae_int32x2 sample_1 = AE_ZERO32();
+	ae_int32x2 sample_2 = AE_ZERO32();
+	ae_valign inu[PLATFORM_MAX_CHANNELS];
+	ae_valign outu = AE_ZALIGN64();
+	uint32_t n, m, nmax, i, j, left, left_samples;
```

Comment:
Have you tried if there's code speed difference with uint32_t vs. int types usage?

Reply:
I haven,t tried, but in PR https://github.com/thesofproject/sof/pull/6024, @lyakh asked to  use uint32_t.

Reply:
I've been wondering what makes better (=faster code), I've used int to let compiler have more freedom when the type is not critical but I have no evidence about speed benefit.

Reply:
@lyakh "The unsigned integer types are ideal for uses that treat storage as a bit array. Using an unsigned
instead of an int to gain one more bit to represent positive integers is almost never a good idea.
Attempts to ensure that some values are positive by declaring variables unsigned will typically be
defeated by the implicit conversion rules (§10.5.1, §10.5.2.1)." from The C++ Programming Language book, section 6.2.4

Reply:
@andrula-song I'm not sure to which my specific comment in that PR you're referring, but I just reviewed what I've written there and I only see there my proposals to use *unsigned* integer types, by which I actually rather meant `unsigned int`, it was your decision then to use `uint32_t` instead, or am I overseeing something? So, yes, I'd use `unsigned int` for unsigned integer types where bit-width isn't critical. But since for the foreseeable future `unsigned int` is the *same* as `uintr32_t` on xtensa, I didn't further object there...
@singalsu I don's think we often use unsigned integers to get the full 32 bit range, maybe for timers, but mostly just to indicate, that that value cannot be negative. Something like 10 years ago I was still unsure what to use for a typical `for (i = 0; i < 10; i++)` but gradually statistics and influence from fellow kernel developers moved me to always use `unsigned int` for those. Since then I almost never get compiler warnings any more about sign mismatch, whereas with `int` in some cases I was getting them.

Reply:
...just to repeat about `(unsigned) int` vs. `(u)int32_t` - I agree with @singalsu - I'd **only** use `(u)int32_t` where it definitely must be a 32-bit variable, not just a generic integer variable with some range guarantees.

Reply:
Thanks for review.
I tried to use xtansa simulator to test the cost of int and unsigned int statement for 500 times, the cylces are the same, so keep like unsignd int?

Reply:
that's expected, that there's no difference. I expect the compiler to generate exactly the same code since `uint32_t` **is** just a typedef of `unsigned int` in all xtensa toolchains so far, I think. So yes, using `unsigned int` here would be my personal preference

Reply:
...you should be able to just compare `.lst` for both cases, the assembly code should be the same

Reply:
One interesting difference between signed and unsigned: signed overflow is undefined behavior:
- https://github.com/zephyrproject-rtos/zephyr/pull/41553
- https://frehberg.com/2022/01/rust-signed-vs-unsigned-arithmetic/

I would recommend putting _readability first_: if the value cannot ever be signed then use `unsigned`. Not to "gain one extra bit" but to _communicate_ better. Same with `uint32_t`: is there any assumption in the code that this value can only be 32 bits? If none then communicate `unsigned`.

In the very rare cases where readability causes performance or correctness issues, then use a more confusing type but with plenty comments to justify why.